src/PAModel/SourceTransforms/AppTestTransform.cs

```diff
@@ -105,16 +105,26 @@ public void AfterRead(BlockNode control)
                 {
                     if (!_screenIdToScreenName.ToDictionary(kvp => kvp.Key, kvp => kvp.Value).TryGetValue(testStep.ScreenId, out var screenName))
                     {
-                        _errors.ValidationError($"ScreenId referenced by TestStep {testStep.Rule} in {control.Name.Identifier} could not be found");
-                        throw new DocumentException();
+                        _errors.ValidationWarning($"ScreenId referenced by TestStep {testStep.Rule} in {control.Name.Identifier} could not be found");
+                        var testStepRuleKey = $"{control.Name.Identifier}.{testStep.Rule}";
+
+                        // checking if this key already exist, not an ideal situation
+                        // Fallback logic just to make sure collisions are avoided.
+                        if (_entropy.RuleScreenIdWithoutScreen.ContainsKey(testStepRuleKey))
+                        {
+                            _errors.ValidationWarning($"RuleScreenIdWithoutScreen has a duplicate key {testStepRuleKey}");
```

Comment:
If we're throwing an exception, should this remain an Error?

Reply:
Yup makes sense, I can change it to an error and remove the names.